---
title: "Validation data Bismark"
author: "Sara Ekberg"
output: html_document
date: '2022-11-07'
---

This document contains some code chunks that has been used to generate plot for the master thesis project **Evaluation and implementation of quality control parameters for genome-wide DNA methylome sequencing**, University of Sk√∂vde HT-2022. Most of the code has been procured from the NBIS Epigenomics Workshop 2022, that the author attended during the projects duration.

The code chunks can not be run without first generating Bismark coverage files and modifying the scripts.

```{r}
library("methylKit")
```

```{r}
# create a the object myobj that contains all the coverage files generated by Bismark, sample id, reference based on the function methRead().

myobj <- methRead(location = list("/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/FU166-GM12878-FU166-FU167-1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/FU166-GM12878-FU166-FU167-2_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SRR13051144_pass_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SRR13051145_pass_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SRR13051146_pass_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SRR13051147_pass_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/MethylSeq-HG001-LAB01-REP01_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/MethylSeq-HG001-LAB01-REP02_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SPLAT-HG001-LAB01-REP01_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz", "/proj/ngi2016001/nobackup/private/workspace_sarek/exjobb/bismark/results/coverage/SPLAT-HG001-LAB01-REP02_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz"),
           sample.id=list("Emseq_FU166-GM12878-FU166-FU167-1", "Emseq_FU166-GM12878-FU166-FU167-2", "Emseq_SRR13051144", "Emseq_SRR13051145", "Emseq-SRR13051146", "Emseq_SRR13051147", "MethylSeq-HG001-LAB01-REP01", "MethylSeq-HG001-LAB01-REP02", "SPLAT-HG001-LAB01-REP01","SPLAT-HG001-LAB01-REP02"),
           pipeline = "bismarkCoverage",
           assembly="GRCh38",
           treatment=c(1,1,1,1,1,1,1,1,1,1),
           mincov = 10
           )
```

```{r}
# Create a histogram of the methylation percentage per sample
# Each sample is listed in the []. Update with number 1-10 for all the samples

getMethylationStats(myobj[[1]], plot=TRUE, both.strands=FALSE)
```

```{r}
# Create a histogram of the read coverage per sample
# Each sample is listed in the [].Update with number 1-10 for all the samples

getCoverageStats(myobj[[1]], plot=TRUE, both.strands=FALSE)
```

```{r}
# Get percentile data by setting plot=FALSE
# Each sample is listed in the [].Update with number 1-10 for all the samples

getCoverageStats(myobj[[1]], plot=FALSE, both.strands=FALSE)

```

```{r}
#Filter out all bases with coverage below 10 and more than 99.9% coverage

myobj.filt <- filterByCoverage(myobj,
                      lo.count=10,
                      lo.perc=NULL,
                      hi.count=NULL,
                      hi.perc=99.9)
```

```{r}
# Normalize the coverage values between samples with a scaling factor derived from differences between the median of the coverage distributions.

myobj.filt.norm <- normalizeCoverage(myobj.filt, method = "median")

```

```{r}
# merge all the samples
meth <- unite(myobj.filt.norm, destrand=FALSE)
```

```{r}
# get percent methylation matrix
pm=percMethylation(meth)
```

```{r}
# calculate standard deviation of CpGs
sds=matrixStats::rowSds(pm)
```

```{r}
# Visualize the distribution of the per-CpG standard deviation
# to determine a suitable cutoff
hist(sds, breaks = 100)
```

```{r}
# keep only CpG with standard deviations larger than 0.5%
meth <- meth[sds > 0.5]
```
